tests/test_backend_connect.py

```diff
@@ -1026,14 +1026,14 @@ def test_connect_pgpass_regular(self):
             # test escaping
             self.run_testcase({
```

Comment:
This testcase was actually incorrect. Backslashes and colons in a postgres:// user/db are treated as literals.

```
psql 'postgres://test\@fgh/test:db?host=/home/matt/.local/share/edgedb/_localdev'
psql: error: connection to server on socket "/home/matt/.local/share/edgedb/_localdev/.s.PGSQL.5432" failed: FATAL:  no pg_hba.conf entry for host "[local]", user "test\", database "test:db", no encryption
```